drivers/amazonec2/amazonec2.go

```diff
@@ -335,10 +349,35 @@ func (d *Driver) Create() error {
 	}
 
 	log.Debugf("launching instance in subnet %s", d.SubnetId)
-	instance, err := d.getClient().RunInstance(d.AMI, d.InstanceType, d.Zone, 1, 1, d.SecurityGroupId, d.KeyName, d.SubnetId, bdm, d.IamInstanceProfile)
-
-	if err != nil {
-		return fmt.Errorf("Error launching instance: %s", err)
+	var instance amz.EC2Instance
+	if d.RequestSpotInstance {
+		spotInstanceRequestId, err := d.getClient().RequestSpotInstances(d.AMI, d.InstanceType, d.Zone, 1, d.SecurityGroupId, d.KeyName, d.SubnetId, bdm, d.IamInstanceProfile, d.SpotPrice)
+		if err != nil {
+			return fmt.Errorf("Error request spot instance: %s", err)
+		}
+		var instanceId string
+		var spotInstanceRequestStatus string
+		log.Infof("Waiting for spot instance fullfillment...")
+		log.Infof("This will take about 3 minutes.")
+		// check util fulfilled
+		for instanceId == "" {
+			time.Sleep(time.Second * 5)
+			spotInstanceRequestStatus, instanceId, err = d.getClient().DescribeSpotInstanceRequests(spotInstanceRequestId)
+			if err != nil {
+				return fmt.Errorf("Error describe spot instance request: %s", err)
+			}
+			log.Infof("Spot instance request status: %s", spotInstanceRequestStatus)
```

Comment:
This should either happen once or pushed into debug.  Otherwise you get a long stream of "pendings".